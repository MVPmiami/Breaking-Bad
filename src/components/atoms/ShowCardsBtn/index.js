import React from "react";
import styles from "./style.module.scss";

const ShowCardsBtn = ({ personPerPage, changeAmountCards, amount }) => {
  return (
    <button
      className={`${styles.amountCards} ${
        personPerPage === amount ? styles.active : ""
      }`}
      onClick={() => {
        changeAmountCards(amount);
      }}
    data-test-showBtn = {`show-${amount}`}>
      {amount}
    </button>
  );
};

export default ShowCardsBtn;
