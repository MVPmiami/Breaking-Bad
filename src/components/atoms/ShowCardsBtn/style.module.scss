@import "./../../../assets/styles/style.scss";

.amountCards {
  min-width: 2rem;
  min-height: 2rem;
  color: $dataText;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  &:hover {
    cursor: pointer;
    color: $CatalogBcgActive;
  }
}
.amountCards.active {
  color: $CatalogBcgActive;
  box-shadow: 0px 0.25rem 0.5rem rgba(0, 0, 0, 0.1);
  border-radius: 0.25rem;
  background: $CatalogBcgDisable;
}
