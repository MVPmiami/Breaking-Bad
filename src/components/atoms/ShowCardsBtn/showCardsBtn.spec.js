import React from "react";
import ShowCardsBtn from "../../../../src/components/atoms/ShowCardsBtn";

const setUp = (props) => shallow(<ShowCardsBtn {...props} />);

describe("should render ShowCardsBtn component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should contain .amountCards wrapper", () => {
    const wrapper = component.find(".amountCards");
    expect(wrapper.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it("pageElement should contain .active class", () => {
    const wrapper = shallow(<ShowCardsBtn personPerPage={1} amount={2} />);
    wrapper.find(".active");
    expect(wrapper.length).toBe(1);
  });

  it("should call callback of amountCards", () => {
    const onBtnClick = jest.fn();
    const wrapper = shallow(<ShowCardsBtn changeAmountCards={onBtnClick} />);
    wrapper.find(".amountCards").simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
  });
});
