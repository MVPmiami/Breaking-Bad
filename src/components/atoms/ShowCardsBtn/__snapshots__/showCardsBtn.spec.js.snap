// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ShowCardsBtn component should contain .amountCards wrapper 1`] = `
<button
  className="amountCards active"
  data-test-showBtn="show-undefined"
  onClick={[Function]}
/>
`;
